src/rust/engine/process_execution/src/remote.rs

```diff
@@ -663,6 +677,20 @@ impl CommandRunner {
             }
             StreamOutcome::StreamClosed(operation_name_opt) => {
               trace!("wait_on_operation_stream (build_id={}) returned stream close, will retry operation_name={:?}", context.build_id, operation_name_opt);
+
+              // Check if the number of request attempts sent thus far have exceeded the number
+              // of retries allowed since the last successful connection. (There is no point in
+              // continually submitting a request if ultimately futile.)
+              if num_retries >= MAX_RETRIES {
+                return Err(
+                  "Too many failures from server, last error was stream close".to_owned(),
```

Comment:
Are you certain that the last error was this issue? If so, is it only the last error, or all the errors were due to the stream being closed?

Grammar tweak:

```python
                  "Too many failures from the server. The last error was from the stream being closed.".to_owned(),
```

Reply:
"last error" is probably bad wording in this case. How about "last event was the server disconnecting with no error given" or similar?

Reply:
Great! Much more descriptive and I don't think risks misleading people that there was an error.

Reply:
Done.